ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/StructHelper.java

```diff
@@ -185,4 +240,47 @@ public static String generateStructMembers(final ImmutableList<StructMember> str
 
         return outputBuilder.toString();
     }
+
+    public static ImmutableSet<C_Type> getArrayStructMemberTypes(final ImmutableList<Struct> allOrderedStructs) {
+        final ImmutableSet.Builder<C_Type> arrayStructMemberTypesBuilder = ImmutableSet.builder();
+        for (final Struct currentStruct : allOrderedStructs) {
+            currentStruct.getStructMembers().stream()
+                    .filter(currentStructMember -> currentStructMember.getType().isArray())
+                    .map(StructMember::getType)
+                    .collect(Collectors.toCollection(HashSet::new))
```

Comment:
Why are you creating a new hashset, to just then turn around and iterate over it.  It seems like at this point you should just call forEach instead of collect then forEach.